File: server/app/schemas/team.py
"""Pydantic schemas for Team and TeamMember."""

from datetime import datetime
from typing import Optional, List
from pydantic import BaseModel, Field


# ── Team schemas ─────────────────────────────────────────────

class TeamBase(BaseModel):
    name: str = Field(..., min_length=1, max_length=100)
    description: str = Field(default="", max_length=500)


class TeamCreate(TeamBase):
    pass


class TeamUpdate(BaseModel):
    name: Optional[str] = Field(default=None, min_length=1, max_length=100)
    description: Optional[str] = Field(default=None, max_length=500)


class TeamMemberAdd(BaseModel):
    user_id: str
    role: str = Field(default="member", pattern="^(member|admin)$")


class TeamMemberUpdate(BaseModel):
    role: str = Field(..., pattern="^(member|admin)$")


# ── Response schemas ─────────────────────────────────────────

class TeamMemberResponse(BaseModel):
    id: str
    user_id: str
    username: str
    email: str
    role: str
    joined_at: datetime

    class Config:
        from_attributes = True


class TeamResponse(TeamBase):
    id: str
    owner_id: str
    owner_username: str
    created_at: datetime
    updated_at: datetime
    member_count: int = 0

    class Config:
        from_attributes = True


class TeamDetailResponse(TeamResponse):
    members: List[TeamMemberResponse] = []

    class Config:
        from_attributes = True


# ── Team Activity / Stats schemas ───────────────────────────

class TeamMemberStatus(BaseModel):
    user_id: str
    username: str
    current_activity: Optional[str] = None
    current_activity_status: Optional[str] = None
    current_activity_started_at: Optional[datetime] = None
    today_duration_seconds: int = 0


class TeamActivityResponse(BaseModel):
    user_id: str
    username: str
    activity_id: str
    activity_name: str
    activity_status: str
    started_at: Optional[datetime] = None
    today_duration_seconds: int = 0

    class Config:
        from_attributes = True

File: apps/web/src/types/team.ts
/** Team-related TypeScript types. */

export interface TeamResponse {
  id: string;
  name: string;
  description: string;
  owner_id: string;
  owner_username: string;
  created_at: string;
  updated_at: string;
  member_count: number;
}

export interface TeamMemberResponse {
  id: string;
  user_id: string;
  username: string;
  email: string;
  role: string;
  joined_at: string;
}

export interface TeamDetailResponse extends TeamResponse {
  members: TeamMemberResponse[];
}

export interface TeamMemberStatus {
  user_id: string;
  username: string;
  current_activity: string | null;
  current_activity_status: string | null;
  current_activity_started_at: string | null;
  today_duration_seconds: number;
}

export interface TeamActivityResponse {
  user_id: string;
  username: string;
  activity_id: string;
  activity_name: string;
  activity_status: string;
  started_at: string | null;
  today_duration_seconds: number;
}

export interface TeamStats {
  team_id: string;
  member_count: number;
  total_activities: number;
  total_duration_seconds: number;
  per_member: {
    user_id: string;
    username: string;
    activity_count: number;
    total_duration_seconds: number;
  }[];
}
